### ISSUE TITLE
fine tune my own dataset?

### ISSUE THREAD

**Issue by HwaiHo-0552** [flyyufelix/cnn_finetune#47]
I really want to know how to load my own dataset for fine tune.
Who could share the code.
Thanks!

**Comment by GreatNewHope** [comment]
No one is going to help you if you don't put enough effort on explaining your type of data and its structure. Do you have your dataset on a pandas dataframe? Is it just a numpy array? You just have some photos and don't know how to convert them to datasets?

**Comment by HwaiHo-0552** [comment]
@margaro95   Yes, I download some images, but I don't know how to convert the dataset to fine tune the densenet.  I try to make dataset like 'flyyufelix', but it was fail. I learn the source code from 'fine tune densenet---- flyyufelix. 
 the NO.218 line---- X_train, Y_train, X_valid, Y_valid=load_cifar10_data(img_rows, img_cols)
X_train, Y_train  come from load_cifar10_data,  and load_cifar10_data come from load_cifar10.py.
so I check the load_cifar10.py        
I found NO.16 line----(X_train, Y_train), (X_valid, Y_valid) = cifar10.load_data()
so I ckeck the API------cifar10.py in keras's API file.
In cifar10.py, the dataset download from 'https://www.cs.toronto.edu/~kriz/cifar-10-python.tar.gz'.
I download the cifar-10-python.tar.gz，which include data_batch_1,  data_batch_2, test_batch,etc
so I plan to make my own dataset like data_batch....., then fit the cifar10.py to convet to the numpy array.
Or, I want to make my own dataset directly like Numpy arrays.
But I dont't know how to convert. 
Thanks!
my dataset is a file: /home/xxx/xxxx/train,   
train includes 200 images that two classes(100 cat.jpg and 100 dog.jpg)

**Comment by GreatNewHope** [comment]
Man, just a quick google search solved the problem... https://stackoverflow.com/questions/7762948/how-to-convert-an-rgb-image-to-numpy-array

```
import cv2
import os
import numpy as np

files = os.listdir()

final_dataset = []

for image in files:
    im = cv2.imread(image)
    flattened_im_list = list(im.flatten())
    final_dataset.append(flattened_im_list)

final_dataset = np.array(final_dataset)
np.save("my_dataset.npy", final_dataset)
```
I haven't tried out this code, I programmed it here in the comment box of github.
It will most probably have problems if your images are not of the same size.
Feel free to comment anything on my code that might not be working

**Comment by HwaiHo-0552** [comment]
@margaro95  Thank you very much

**Comment by lizirui287** [comment]
> @ margaro95 是的，我下载了一些图片，但我不知道如何转换数据集以微调密集网络。我尝试制作像'flyyufelix'这样的数据集，但它失败了。我从'微调密网 - flyyufelix中学习源代码。
> NO.218行---- X_train，Y_train，X_valid，Y_valid = load_cifar10_data（img_rows，img_cols）
> X_train，Y_train来自load_cifar10_data，load_cifar10_data来自load_cifar10.py。
> 所以我检查了load_cifar10.py 
> 我发现NO.16行----（X_train，Y_train），（X_valid，Y_valid）= cifar10.load_data（）
> 所以我在keras的ckeck API------cifar10.py API文件。
> 在cifar10.py中，从“ https://www.cs.toronto.edu/~kriz/cifar-10-python.tar.gz ” 下载数据集。
> 我下载了cifar-10-python.tar.gz，其中包括data_batch_1，data_batch_2，test_batch等，
> 所以我打算创建自己的数据集，如data_batch .....，然后将cifar10.py调整到numpy数组。
> 或者，我想像Numpy数组一样直接创建自己的数据集。
> 但我不知道如何转换。
> 谢谢！
> 我的数据集是一个文件：/ home / xxx / xxxx / train，
> train包括200个图像，两个类（100 cat.jpg和100 dog.jpg）

Hey, I have the same problem with you now, is your problem solved? Is there a solution? Would you like to share it?

**Comment by digitalized-hfut** [comment]
You could follow the way like margaro95 that load your own dataset based on python library(numpy, os and opencv). Before to finish that, you might learn some basic knowledge about numpy, os and opencv.      
But, I recommend you check the API of Keras(Image Processing). It is a easy way that process the local dataset.

**Comment by HwaiHo-0552** [comment]
You could follow the way like margaro95 that process your own dataset based on python library(numpy, os and opencv). Before to finish that, you might learn some basic knowledge about numpy, os and opencv.
But, I recommend you check the API of Keras(Image Processing). It is a easy way that process the local dataset.
There is another way to finish the fine-tune or transfer work. Pytorch is a good idea.
可以按照 margaro95的方法进行处理图像数据，但是前提是你得了解到python对数据的处理机制，而且你也得知道numpy、os、opencv库的使用和机制，难度倒不是很大，但是要花一些时间去学习基本的知识。
如果你想快速使用，那就去keras的文档查，Image Processing封装的API使用，这个大概学学，你去改一点点代码，应该比较容易，但是我觉得keras封装的太好了，很多东西看不到，所以不利于理解，这仅是我个人的观点。
所以，我还建议，如果你tensorflow玩不转，搞不懂，那就上pytorch，封装的没有keras那么高阶，也没有tensorflow那么偏低层，对于编程能力不好的，pytorch不仅能让你上手舒服，而且还能一点点学习下代码。

> > @ margaro95 是的，我下载了一些图片，但我不知道如何转换数据集以微调密集网络。我尝试制作像'flyyufelix'这样的数据集，但它失败了。我从'微调密网 - flyyufelix中学习源代码。
> > NO.218行---- X_train，Y_train，X_valid，Y_valid = load_cifar10_data（img_rows，img_cols）
> > X_train，Y_train来自load_cifar10_data，load_cifar10_data来自load_cifar10.py。
> > 所以我检查了load_cifar10.py
> > 我发现NO.16行----（X_train，Y_train），（X_valid，Y_valid）= cifar10.load_data（）
> > 所以我在keras的ckeck API------cifar10.py API文件。
> > 在cifar10.py中，从“ https://www.cs.toronto.edu/~kriz/cifar-10-python.tar.gz ” 下载数据集。
> > 我下载了cifar-10-python.tar.gz，其中包括data_batch_1，data_batch_2，test_batch等，
> > 所以我打算创建自己的数据集，如data_batch .....，然后将cifar10.py调整到numpy数组。
> > 或者，我想像Numpy数组一样直接创建自己的数据集。
> > 但我不知道如何转换。
> > 谢谢！
> > 我的数据集是一个文件：/ home / xxx / xxxx / train，
> > train包括200个图像，两个类（100 cat.jpg和100 dog.jpg）
> 
> Hey, I have the same problem with you now, is your problem solved? Is there a solution? Would you like to share it?
